homeassistant/components/sensor/imap.py

```diff
@@ -0,0 +1,104 @@
+"""
+Support for Google travel time sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.google_travel_time/
+"""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:email-outline'
+
+CONF_USER = "user"
+CONF_PASSWORD = "password"
+CONF_SERVER = "server"
+CONF_PORT = "port"
+CONF_NAME = "name"
+CONF_PERIOD = "frequency"
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required('platform'): 'imap',
+    vol.Optional(CONF_NAME): vol.Coerce(str),
+    vol.Required(CONF_USER): vol.Coerce(str),
+    vol.Required(CONF_PASSWORD): vol.Coerce(str),
+    vol.Required(CONF_SERVER): vol.Coerce(str),
+    vol.Optional(CONF_PORT): vol.Coerce(int),
```

Comment:
Add a range, [like this](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mqtt/__init__.py#L93-L94).